pyanaconda/storage/checker.py

```diff
@@ -411,6 +412,51 @@ def verify_mounted_partitions(storage, constraints, report_error, report_warning
                                "installation. Please unmount it and retry.") % part.path)
 
 
+def verify_lvm_destruction(storage, constraints, report_error, report_warning):
+    """ Verify that destruction of LVM devices is correct.
+
+    This means: If a PV is destroyed, the VG on it must be destroyed too.
+    Additionally, destroying an incomplete VG warns about this.
+
+    :param storage: a storage to check
+    :param constraints: a dictionary of constraints
+    :param report_error: a function for error reporting
+    :param report_warning: a function for warning reporting
+    """
+    # Implementation of the verification uses an inverted approach: Check that all PVs to be
+    # destroyed are contained in a VG that will be destroyed too. If there is a PV to be destroyed
+    # that has a VG that isn't going to be destroyed, that is an error. This is because we ask
+    # blivet to ignore some disks, and these might contain a PV. Crucially, blivet implements this
+    # such that the ignoring hides also all dependent devices - for any disk, it could be a PV
+    # partition, which would then hide also all VGs and LVs that depend on the PV. It does not
+    # matter that the VG needs other PVs too - the hiding wins.
+    log.debug("called verify_lvm_destruction()")
+    destroyed_vg_names = set()
+    vg_names_seen_in_destroyed_pvs = set()
+    for action in storage.devicetree.actions:
+        if isinstance(action, ActionDestroyDevice) and action.device.type == "lvmvg":
+            log.debug("ActionDestroyDevice: %s %s", action.device.type, action.device.name)
+            destroyed_vg_names.add(action.device.name)
+            if not action.device.complete:
+                report_warning("Disk selection will result in removing an incomplete volume "
+                               "group '{}'.".format(action.device.name))
+        elif isinstance(action, ActionDestroyFormat) and action.orig_format.type == "lvmpv":
+            log.debug("ActionDestroyFormat: %s %s", action.orig_format.type, action.device.name)
+            # Must check if the PV actually has a VG assigned
+            if action.orig_format.vg_name:
+                vg_names_seen_in_destroyed_pvs.add(action.orig_format.vg_name)
+    missing_vg_names = vg_names_seen_in_destroyed_pvs - destroyed_vg_names
+    log.debug("destroyed VGs: %s", destroyed_vg_names)
+    log.debug("seen VGs: %s", vg_names_seen_in_destroyed_pvs)
+    log.debug("missing VGs: %s", missing_vg_names)
+    if missing_vg_names:
+        for vg_name in missing_vg_names:
+            report_error(_("Disk selection contains only some physical volumes for volume group "
+                           "'{}'. Adjust your 'ignoredisk' command, or remove the physical "
```

Comment:
This validation runs for all types of partitioning, so these messages should be generic. I would recommend to ask users to adjust their disk selection, or remove the physical volumes. In some cases, it will be possible to solve this validation error by changing the disk selection in the storage spoke.

Reply:
First - by generic, you mean they should not mention kickstart, right?

Second - Does it make sense to see the LVM details here, or do you think walking the changes to speak only about disks is better? The error _is_ caused specifically by LVM stuff, but if I changed it to list disks, I could reuse the existing error message from GUI.

Reply:
> First - by generic, you mean they should not mention kickstart, right?

Yes, they should not mention kickstart. I would replace only `Adjust your 'ignoredisk' command` with `Adjust your disk selection`. Otherwise, the message looks good to me.

> Second - Does it make sense to see the LVM details here, or do you think walking the changes to speak only about disks is better? The error _is_ caused specifically by LVM stuff, but if I changed it to list disks, I could reuse the existing error message from GUI.

 It is definitely good to mention the cause of the error and provide some suggestions how to fix it or avoid it, so the LVM details should be part of the message.

Reply:
Done.